src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Nullable.cs

```diff
@@ -1007,6 +1101,231 @@ static void M()
                 Diagnostic(ErrorCode.ERR_PredefinedTypeNotFound, "object? x").WithArguments("System.Runtime.CompilerServices.NullableAttribute").WithLocation(5, 16));
         }
 
+        [Fact]
+        public void Tuples()
+        {
+            var source =
+@"public class A
+{
+    public static ((object?, object) _1, object? _2, object _3, ((object?[], object), object?) _4) Nested;
+    public static (object? _1, object _2, object? _3, object _4, object? _5, object _6, object? _7, object _8, object? _9) Long;
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular8);
+            CompileAndVerify(comp, validator: assembly =>
+            {
+                var reader = assembly.GetMetadataReader();
+                var typeDef = GetTypeDefinitionByName(reader, "A");
+                var fieldDefs = typeDef.GetFields().Select(f => reader.GetFieldDefinition(f)).ToArray();
+
+                // Nested tuple
+                var field = fieldDefs.Single(f => reader.StringComparer.Equals(f.Name, "Nested"));
+                var customAttributes = field.GetCustomAttributes();
+                AssertAttributes(reader, customAttributes,
+                    "MemberReference:Void System.Runtime.CompilerServices.TupleElementNamesAttribute..ctor(String[])",
+                    "MethodDefinition:Void System.Runtime.CompilerServices.NullableAttribute..ctor(Boolean[])");
+                var customAttribute = reader.GetCustomAttribute(customAttributes.ElementAt(1));
+                AssertEx.Equal(ImmutableArray.Create(false, false, true, false, true, false, false, false, false, true, false, true), reader.ReadBoolArray(customAttribute.Value));
```

Comment:
> AssertEx.Equal(ImmutableArray.Create(false, false, true, false, true, false, false, false, false, true, false, true), reader.ReadBoolArray(customAttribute.Value)); [](start = 15, length = 164)

I am not sure why are we going that low level in our validation when we should always be able to get to the symbols for the produced module and assert that we have question marks in all the right places. I think ToTestDisplayString would be good enough and human readable. I cannot verify whether this assert is correct and whether other aspects (like tuple element names) are properly preserved in reasonable amount of time.